week-01/seonju-oauth2.md

```diff
@@ -0,0 +1,94 @@
+# OAuth 2.0
+
+## What is OAuth 2.0?
+
+OAuth 2.0, which stands for “Open Authorization”, is a standard designed to allow a website or application to access resources hosted by other web apps on behalf of a user. OAuth 2.0 provides consented access and restricts actions of what the client app can perform on resources on behalf of the user, without ever sharing the user's credentials.
+
+## **How Does OAuth 2.0 Work?**
+
+### **Roles**
+
+Before we take a look at how it works, let's take a look at the roles specified as standards in OAuth2.0.
+
+These define the essential components of an OAuth 2.0 system, and are as follows:
+
+- **Resource Owner** is the user or system that owns the protected resources and can grant access to them.
+- The **Client** is the system that requires access to the protected resources. To access resources, the Client must hold the appropriate Access Token.
+- **Authorization Server** receives requests from the Client for Access Tokens and issues them upon successful authentication and consent by the Resource Owner. The authorization server exposes two endpoints: the Authorization endpoint, which handles the interactive authentication and consent of the user, and the Token endpoint, which is involved in a machine to machine interaction.
+- **Resource Server** that protects the user’s resources and receives access requests from the Client. It accepts and validates an Access Token from the Client and returns the appropriate resources to it.
+
+### OAuth 2.0 Protocol Flow
+
```

Comment:
The examples you provided are beneficial in illustrating how OAuth works in practice!